src/cc/lib/distributed/graph_engine.cc

```diff
@@ -110,13 +118,25 @@ grpc::Status GraphEngineServiceImpl::GetNodeFeatures(::grpc::ServerContext *cont
             continue;
         }
 
-        response->add_offsets(node_offset);
         response->mutable_feature_values()->resize(feature_offset + fv_size);
```

Comment:
Avoid multiple resizes, here and on line 138. Append features if and only if they are present in the partition.

Reply:
We need to resize it beforehand so if the feature does exist, it'll fit. We only do the second resize if the feature is not found. This is a copy of the GetEdgeFeatures code which also has 2.